compiler/rustc_monomorphize/src/collector.rs

```diff
@@ -651,6 +662,38 @@ impl<'a, 'tcx> MirUsedCollector<'a, 'tcx> {
         );
         self.move_size_spans.push(source_info.span);
     }
+
+    fn check_fn_args_move_size(
+        &mut self,
+        callee_ty: Ty<'tcx>,
+        args: &[mir::Operand<'tcx>],
+        location: Location,
+    ) {
+        let limit = self.tcx.move_size_limit();
+        if limit.0 == 0 {
+            return;
+        }
+
+        if args.is_empty() {
+            return;
+        }
+
+        // Allow large moves into container types that themselves are cheap to move
+        let ty::FnDef(def_id, _) = *callee_ty.kind() else {
+            return;
+        };
+        if self
+            .skip_move_check_fns
+            .get_or_insert_with(|| build_skip_move_check_fns(self.tcx))
+            .contains(&def_id)
+        {
+            return;
+        }
+
+        for arg in args {
+            self.check_operand_move_size(arg, location);
```

Comment:
If this is `Operand::Move`, should the lint really fire? At least in some cases those don't cause an actual `memcpy` so there's no reason to lint against them. (But that only applies to function arguments! When used in `Rvalue`, they do still incur a `memcpy`.)

Reply:
That is indeed something we need to take into account and that I plan on doing later. I added a task
> Differentiate between `Operand::Move` and `Operand::Copy`

about it to the [tracking issue](https://github.com/rust-lang/rust/issues/83518) now for improved visibility.